src/components/EditContentModal/EditContentModal.tsx

```diff
@@ -0,0 +1,445 @@
+
+  const closeModal = () => {
+    setClosed();
+    formik.resetForm();
+  };
+
+  const { mutateAsync: editContent, isLoading: isEditing } = useEditContentQuery(
+    queryClient,
+    mapFormikToEditAPIValues(formik.values),
+  );
+
+  const createDataLengthOf1 = formik.values.length === 1;
+
+  const allExpanded = !formik.values.some(({ expanded }) => !expanded);
+
+  const expandAllToggle = () => {
+    formik.setValues([...formik.values.map((vals) => ({ ...vals, expanded: !allExpanded }))]);
+  };
+
+  const updateVariable = (index: number, newValue) => {
+    const updatedData = [...formik.values];
+    updatedData[index] = { ...updatedData[index], ...newValue };
+    formik.setValues(updatedData);
+  };
+
+  const getFieldValidation = (
+    index: number,
+    field: keyof FormikEditValues,
+  ): 'default' | 'success' | 'error' => {
+    const hasNotChanged = isEqual(initialValues[index]?.[field], formik.values[index]?.[field]);
+    const errors = !!formik.errors[index]?.[field];
+    switch (true) {
+      case errors:
+        return 'error';
+      case hasNotChanged:
+        return 'default';
+      case !hasNotChanged:
+        return 'success';
+      default:
+        return 'default';
+    }
+  };
+
+  const debouncedValues = useDebounce(formik.values);
+
+  const { mutateAsync: validateContentList } = useValidateContentList();
+
+  useEffect(() => {
+    if (open)
+      validateContentList(debouncedValues.map(({ name, url, uuid }) => ({ name, url, uuid }))).then(
+        async (validationData) => {
+          const formikErrors = await formik.validateForm(debouncedValues);
+          const mappedErrorData = mapValidationData(validationData, formikErrors);
+          formik.setErrors(mappedErrorData);
+        },
+      );
+  }, [debouncedValues, values, open]);
+
+  const onToggle = (index: number) => {
+    if (formik.values[index]?.expanded) {
+      updateVariable(index, { ...formik.values[index], expanded: false });
+    } else updateVariable(index, { ...formik.values[index], expanded: true });
+  };
+
+  const updateArchAndVersion = (index: number) => {
+    const url = formik.values[index]?.url;
+    if (isValidURL(url)) {
+      const arch =
+        formik.values[index]?.arch ||
+        distArches.find(({ name, label }) => url.includes(name) || url.includes(label))?.label ||
+        '';
+
+      let versions: Array<string> = [];
+      if (formik.values[index]?.versions?.length) {
+        versions = formik.values[index]?.versions;
+      } else {
+        const newVersion = distVersions.find(
+          ({ name, label }) => url.includes(name) || url.includes(label),
+        )?.label;
+        if (newVersion) versions = [newVersion];
+      }
+      updateVariable(index, { arch, versions });
+    }
+  };
+
+  const urlOnBlur = (index: number) => {
+    updateArchAndVersion(index);
+  };
+
+  return (
+    <Modal
+      variant={ModalVariant.medium}
+      title='Edit Custom Content'
```

Comment:
This says "Custom Content", everything else says "Content Source". Is that intentional?

Reply:
nope